Source/missiles.cpp

```diff
@@ -3992,7 +3990,7 @@ void MI_Lightctrl(int i)
 		missile[i]._miVar1 = missile[i].position.tile.x;
 		missile[i]._miVar2 = missile[i].position.tile.y;
 	}
-	if (missile[i]._mirange == 0 || mx == 0 || my == 0) {
+	if (missile[i]._mirange == 0) {

```

Comment:
levels always have a 16 tile border so it shouldn't be able to reach 0,0 to begin with, not that this was original checking for the full map, but was reduced in https://github.com/diasurgical/devilutionX/pull/2115 as most of the check was logically impossible as well.

Reply:
Could you add asserts at line 3942 checking that mx/my are >= 0? This will match the existing checks for missiles outside the dungeon upper limits and they get compiled out of non debug builds anyway.

Reply:
Done